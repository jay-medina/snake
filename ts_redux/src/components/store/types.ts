import {
  MapDispatchToProps as MDTP,
  MapStateToProps as MSTP,
} from 'react-redux';
import { AppState } from './util';

export type MapDispatchToProps<DispatchProps, OwnProps = {}> = MDTP<
  DispatchProps,
  OwnProps
>;

export type MapStateToProps<StateProps, OwnProps = {}> = MSTP<
  StateProps,
  OwnProps,
  AppState
>;
