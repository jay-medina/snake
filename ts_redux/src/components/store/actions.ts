// import { GameThunkAction } from './types';
// import { AppState } from './util';

interface GameAction {
  type: 'START_GAME';
}

export type AppAction = GameAction;

export const startGame = (): GameAction => ({
  type: 'START_GAME',
});

// export function randomizeApple(appState: AppState) {

// }

// export const startGameThunk = (): GameThunkAction => (dispatch) => {
//   // randomize the apple position
//   // start the game
// //  return dispatch(startGame);
// };
