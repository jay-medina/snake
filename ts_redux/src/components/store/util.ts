export enum GameState {
  Start,
  Run,
  GameOver,
}

export interface AppState {
  gameState: GameState;
}

export const initialState: AppState = {
  gameState: GameState.Start,
};
