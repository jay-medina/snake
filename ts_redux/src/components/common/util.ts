import { GridItem, Snake } from 'store/util';

const isAtPosition = (gridItem1: GridItem) => (gridItem2: GridItem) => {
  return gridItem1.row === gridItem2.row && gridItem1.col === gridItem2.col;
};

export const isSnakeAtPosition = (snake: Snake) => (gridItem: GridItem) =>
  !!snake.find(isAtPosition(gridItem));

export const isTheApple = isAtPosition;
