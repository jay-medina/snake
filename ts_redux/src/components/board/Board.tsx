import React from 'react';
import { Apple, Snake } from '../../store/util';
import { Row } from './Row';
import { AppState } from '../../store/util';
import { connect } from 'react-redux';

interface StateProps {
  rows: number;
  cols: number;
  apple: Apple;
  snake: Snake;
}

type BoardProps = StateProps;

export const Board: React.SFC<BoardProps> = ({ rows, cols, snake, apple }) => {
  const components = [];

  for (let r = 0; r < rows; r += 1) {
    components.push(<Row key={r} cols={cols} row={r} snake={snake} apple={apple} />);
  }

  return <div className="snake__board">{components}</div>;
};

Board.displayName = 'Board';

const mapStateToProps = (state: AppState): StateProps => {
  const { dimensions, snake } = state;

  return {
    apple: state.apple,
    rows: dimensions.rows,
    cols: dimensions.cols,
    snake,
  };
};

export default connect(mapStateToProps)(Board);
