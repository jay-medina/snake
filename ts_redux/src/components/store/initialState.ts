import { AppState } from "./util";


const loadInitialState = (): Partial<AppState> => {
  return {
    apple: 
  };
}
