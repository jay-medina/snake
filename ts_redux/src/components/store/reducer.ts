import { AppAction } from './actions';
import { AppState, GameState, initialState } from './util';

type Game = (start: AppState | undefined, action: AppAction) => AppState;

export const game: Game = (state = initialState, action) => {
  if (action.type === 'START_GAME') {
    return {
      gameState: GameState.Run,
    };
  }

  return state;
};
