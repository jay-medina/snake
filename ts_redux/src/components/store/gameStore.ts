import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { game } from './reducer';

export function createGameStore() {
  return createStore(game, applyMiddleware(thunk));
}
