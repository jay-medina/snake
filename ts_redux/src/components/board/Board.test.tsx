import { mount } from 'enzyme';
import React from 'react';
import Board from './Board';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { AppState, GameState } from '../../store/util';

describe('<Board />', () => {
  it('renders a 5x5 board', () => {
    const mockStore = configureStore<AppState>();
    const store = mockStore({
      gameState: GameState.Start,
      apple: {
        row: 3,
        col: 3,
      },
      dimensions: {
        rows: 5,
        cols: 5,
      },
    });

    const component = mount(
      <Provider store={store}>
        <Board />
      </Provider>,
    );
    expect(component.find('Board')).toMatchSnapshot();
  });
});
