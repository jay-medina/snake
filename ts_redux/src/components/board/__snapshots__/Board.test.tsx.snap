// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> renders a 5x5 board 1`] = `
<Board
  apple={
    Object {
      "col": 3,
      "row": 3,
    }
  }
  cols={5}
  rows={5}
  snake={
    Array [
      Object {
        "col": 1,
        "row": 1,
      },
    ]
  }
>
  <div
    className="snake__board"
  >
    <Row
      apple={
        Object {
          "col": 3,
          "row": 3,
        }
      }
      cols={5}
      key="0"
      row={0}
      snake={
        Array [
          Object {
            "col": 1,
            "row": 1,
          },
        ]
      }
    >
      <div
        className="snake__board-row"
      >
        <Col
          key="0"
        >
          <div
            className="snake__board-col"
          />
        </Col>
        <Col
          key="1"
        >
          <div
            className="snake__board-col"
          />
        </Col>
        <Col
          key="2"
        >
          <div
            className="snake__board-col"
          />
        </Col>
        <Col
          key="3"
        >
          <div
            className="snake__board-col"
          />
        </Col>
        <Col
          key="4"
        >
          <div
            className="snake__board-col"
          />
        </Col>
      </div>
    </Row>
    <Row
      apple={
        Object {
          "col": 3,
          "row": 3,
        }
      }
      cols={5}
      key="1"
      row={1}
      snake={
        Array [
          Object {
            "col": 1,
            "row": 1,
          },
        ]
      }
    >
      <div
        className="snake__board-row"
      >
        <Col
          key="0"
        >
          <div
            className="snake__board-col"
          />
        </Col>
        <Col
          filled="snake"
          key="1"
        >
          <div
            className="snake__board-col snake__board-col-snake"
          />
        </Col>
        <Col
          key="2"
        >
          <div
            className="snake__board-col"
          />
        </Col>
        <Col
          key="3"
        >
          <div
            className="snake__board-col"
          />
        </Col>
        <Col
          key="4"
        >
          <div
            className="snake__board-col"
          />
        </Col>
      </div>
    </Row>
    <Row
      apple={
        Object {
          "col": 3,
          "row": 3,
        }
      }
      cols={5}
      key="2"
      row={2}
      snake={
        Array [
          Object {
            "col": 1,
            "row": 1,
          },
        ]
      }
    >
      <div
        className="snake__board-row"
      >
        <Col
          key="0"
        >
          <div
            className="snake__board-col"
          />
        </Col>
        <Col
          key="1"
        >
          <div
            className="snake__board-col"
          />
        </Col>
        <Col
          key="2"
        >
          <div
            className="snake__board-col"
          />
        </Col>
        <Col
          key="3"
        >
          <div
            className="snake__board-col"
          />
        </Col>
        <Col
          key="4"
        >
          <div
            className="snake__board-col"
          />
        </Col>
      </div>
    </Row>
    <Row
      apple={
        Object {
          "col": 3,
          "row": 3,
        }
      }
      cols={5}
      key="3"
      row={3}
      snake={
        Array [
          Object {
            "col": 1,
            "row": 1,
          },
        ]
      }
    >
      <div
        className="snake__board-row"
      >
        <Col
          key="0"
        >
          <div
            className="snake__board-col"
          />
        </Col>
        <Col
          key="1"
        >
          <div
            className="snake__board-col"
          />
        </Col>
        <Col
          key="2"
        >
          <div
            className="snake__board-col"
          />
        </Col>
        <Col
          filled="apple"
          key="3"
        >
          <div
            className="snake__board-col snake__board-col-apple"
          />
        </Col>
        <Col
          key="4"
        >
          <div
            className="snake__board-col"
          />
        </Col>
      </div>
    </Row>
    <Row
      apple={
        Object {
          "col": 3,
          "row": 3,
        }
      }
      cols={5}
      key="4"
      row={4}
      snake={
        Array [
          Object {
            "col": 1,
            "row": 1,
          },
        ]
      }
    >
      <div
        className="snake__board-row"
      >
        <Col
          key="0"
        >
          <div
            className="snake__board-col"
          />
        </Col>
        <Col
          key="1"
        >
          <div
            className="snake__board-col"
          />
        </Col>
        <Col
          key="2"
        >
          <div
            className="snake__board-col"
          />
        </Col>
        <Col
          key="3"
        >
          <div
            className="snake__board-col"
          />
        </Col>
        <Col
          key="4"
        >
          <div
            className="snake__board-col"
          />
        </Col>
      </div>
    </Row>
  </div>
</Board>
`;
